common/types/loadbalancer.go

```diff
@@ -264,6 +281,26 @@ func NewL3n4AddrFromBackendModel(base *models.BackendAddress) (*L3n4Addr, error)
 		return nil, fmt.Errorf("Invalid IP address \"%s\"", *base.IP)
 	}
 
+	return &LBBackEnd{
+		L3n4Addr: L3n4Addr{IP: ip, L4Addr: *l4addr},
+		Weight:   base.Weight,
+	}, nil
+}
+
+func NewL3n4AddrFromBackendModel(base *models.BackendAddress) (*L3n4Addr, error) {
+	if base.IP == nil {
+		return nil, fmt.Errorf("Missing IP address")
+	}
+
+	l4addr, err := NewL4Addr(NONE, base.Port)
```

Comment:
Can you add a FIXME in the `NewLBBackEndFromBackendModel` and here? after that I think everything is good ;)